### ISSUE TITLE
[BUG]: [skip ci] doesn't prevent automatic build validation pipelines from triggering.

### ISSUE THREAD

**Issue by shengdoescoding** [microsoft/azure-pipelines-agent#4576]
### What happened?

As per the documentation found here: https://learn.microsoft.com/en-us/azure/devops/pipelines/repos/azure-repos-git?view=azure-devops&tabs=yaml&WT.mc_id=DOP-MVP-5001511#pr-triggers

According to this:
"The pipelines triggered by changes to the PR's source branch, if there are no pushed commits whose messages or descriptions contain [skip ci] (or any of its variants). If at least one pushed commit contains [skip ci], the pipelines will not run."

I should be able to have a PR open, make changes to the source branch, commit with [skip ci] in the message, and push the changes to remote without triggering the Build Validation Pipeline.

However, when I attempt this, with an open PR, make changes on local source branch, commit with `git commit -m '[skip ci]'`, then push this change to remote, the build validation pipeline is still triggered.

In the build validation pipeline, I have specified `trigger: none`.

### Versions

Current agent version: '3.232.0'
Operating System
Microsoft Windows Server 2019

### Environment type (Please select at least one enviroment where you face this issue)

- [ ] Self-Hosted
- [X] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Windows 10 Enterprise

### Version controll system

Azure Git

**Comment by DmitriiBobreshev** [comment]
Hi @shengdoescoding, thank you for the feedback, but the agent is not responsible for the PR triggers, could you please create an issue in the [developercommunity](https://developercommunity.visualstudio.com/AzureDevOps)?
I'm closing your ticket as we can't help you, but feel free to reopen it if you disagree.

**Comment by shengdoescoding** [comment]
Hi @DmitriiBobreshev just confirming what you mean. The agent is only responsible for CI triggers?

I thought this was the right place to post the bug report becasue I saw other issues like #858 talking about "skip ci", and other feature requests that has been merged already for additional spellings for "skip ci".

So, although the documentation I linked above also uses "skip ci", it is not the same as the ones mentioned in #858 because the pipeline trigger is a build validation rather than a `git push` trigger?
